services/bettersocial/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,20 @@
+const {messaging} = require('firebase-admin');
+const FcmTokenFunction = require('../../../databases/functions/fcmToken');
+const {FcmToken, User} = require('../../../databases/models');
+const UsersFunction = require('../../../databases/functions/users');
+
+const sendChatNotification = async (userTargetId, payload) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
+  const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, signedUser);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `findSignedUserId` and `findTokenByUserId` functions are being awaited sequentially. These two operations could be performed in parallel using `Promise.all` to improve performance.
```diff
-  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
-  const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, signedUser);
+  const [signedUser, userTargetToken] = await Promise.all([
+    UsersFunction.findSignedUserId(User, userTargetId),
+    FcmTokenFunction.findTokenByUserId(FcmToken, userTargetId)
+  ]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->